api/hypershift/v1beta1/azure_types.go

```diff
@@ -0,0 +1,213 @@
+package v1beta1
+
+// AzureVMImageType is used to specify the source of the Azure VM boot image.
+// Valid values are ImageID and AzureMarketplace.
+// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
+type AzureVMImageType string
+
+const (
+	// ImageID is the used to specify that an Azure resource ID of a VHD image is used to boot the Azure VMs from.
+	ImageID AzureVMImageType = "ImageID"
+
+	// AzureMarketplace is used to specify the Azure Marketplace image info to use to boot the Azure VMs from.
+	AzureMarketplace AzureVMImageType = "AzureMarketplace"
+)
+
+// AzureNodePoolPlatform is the platform specific configuration for an Azure node pool.
+// +kubebuilder:validation:XValidation:rule="!has(self.diskStorageAccountType) || self.diskStorageAccountType != 'UltraSSD' || self.diskStorageAccountType <= 32,767",message=""
+type AzureNodePoolPlatform struct {
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	//
+	// TODO: This should be validated with a regex, based on https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
+	// +kubebuilder:validation:Required
+	VMSize string `json:"vmSize"`
+
+	// image is used to configure the VM boot image. If unset, the default image at the location below will be used and
+	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
+	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
+	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
+	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
+	// +kubebuilder:validation:Required
+	Image AzureVMImage `json:"image"`
+
+	// diskSizeGiB is the size in GiB (1024^3 bytes) to assign to the OS disk. This should be between 16 and 65,536.
+	// When not set, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is 30.
```

Comment:
Does this need to be removed?

Reply:
Or maybe updated to say CAPZ will default this to 30.

Reply:
No, typically we still say the default, it aligns with the previous sentence, and then the nodepool controller should be enforcing this default